src/plugin/src/main/java/com/defenseunicorns/uds/keycloak/plugin/clientpolicy/executor/UDSClientPolicyPermissionsExecutor.java

```diff
@@ -82,13 +137,58 @@ boolean isOwnedByUDSOperator(ClientModel client) {
         return ATTRIBUTE_UDS_OPERATOR_VALUE.equals(client.getAttribute(ATTRIBUTE_UDS_OPERATOR));
     }
 
+    void setAllowedProtocolMappers(ClientRepresentation rep) {
+        if (rep.getProtocolMappers() != null && !rep.getProtocolMappers().isEmpty()) {
+            rep.getProtocolMappers()
+                    .removeIf(mapper -> !configuration.getAllowedProtocolMappers().contains(mapper.getProtocolMapper()));
+        }
+    }
+
+    void setAllowedCustomClientScopes(ClientRepresentation rep) {
+        if (rep.getDefaultClientScopes() != null && !rep.getDefaultClientScopes().isEmpty()) {
+            rep.getDefaultClientScopes()
+                    .removeIf(scope -> !configuration.getAllowedClientScopes().contains(scope));
```

Comment:
We probably need an extra check for `configuration.getAllowedClientScopes()` to keep operating in case of misconfiguration or Keycloak SPI changes. I'll add it along with other comments.

Reply:
Done